go/test/endtoend/backup/vtctlbackup/backup_utils.go

```diff
@@ -812,6 +817,44 @@ func checkTabletType(t *testing.T, alias string, tabletType topodata.TabletType)
 	require.Failf(t, "checkTabletType failed.", "Tablet type is not correct. Expected: %v", tabletType)
 }
 
+func doNotDemoteNewlyPromotedPrimaryIfReparentingDuringBackup(t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	// Start the backup on a replica
+	go func() {
+		// ensure this is a primary first
+		checkTabletType(t, primary.Alias, topodata.TabletType_PRIMARY)
+
+		// now backup
+		err := localCluster.VtctlclientProcess.ExecuteCommand("Backup", replica1.Alias)
+		require.Nil(t, err)
+		wg.Done()
```

Comment:
nit: please `defer wg.Done()` at the beginning of the function

Reply:
done